mlos_bench/mlos_bench/launcher.py

```diff
@@ -142,3 +147,11 @@ def load_env(self) -> Environment:
         return self._config_loader.load_environment(
             self._env_config_file, self._global_config,
             service=LocalExecService(parent=self._config_loader))
+
+    def load_generic(self, env: Environment, cls, json_file_name: str):
```

Comment:
Prefer the old method of `new()` and `load()` in part due to type hint check ability I think.

Reply:
Though, if `cls` is actually a type (it's not clear from this), but below it looks to be.

Then, I think we can use `TypeVar` tricks to make this look like it returns a bound-class/generic so that the passed in var is also the type it needs to return.

Reply:
I agree, I am not quite happy with that `.load_generic()` method myself. OTOH, defining `.load()` and `.new()` in every loadable class is a lot of code duplication, and it does not solve the root problem with the types, only pushes it deeper down into the `util.instantiate_from_config()` function.

I really like the `TypeVar` idea, though - let me play with it and I'll report to you later

Reply:
I think I've been overengineering it. Let's just have two functions, `.load_optimizer()` and `.load_storage()` - I don't think we'll have any other major subsystems in addition to our existing `Environment` / `Optimizer` / `Storage` trio anyway.